### PR TITLE
[CARBONDATA-574] Added thrift server support to Spark 2.0 integration

### PR DESCRIPTION
**Pull request by ravipesala**
Added thrift server support to Spark 2.0 integration

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/349/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  def main(args: Array[String]): Unit = {
+
+    import org.apache.spark.sql.CarbonSession._
+
+    val sparkConf = new SparkConf(loadDefaults = true)
+    val builder = SparkSession
+      .builder()
+      .config(sparkConf)
+      .appName("Carbon Thrift Server")
```
Comment (chenliang613) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: suggest changing appName to : "Carbon Thrift Server for Spark2.x" to distinguish with Spark1.x

Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: I think it should better be "ThriftServer based on CarbonSession"

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    val spark = builder.getOrCreateCarbonSession()
+    val warmUpTime = CarbonProperties.getInstance().getProperty("carbon.spark.warmUpTime", "5000")
+    try {
+      Thread.sleep(Integer.parseInt(warmUpTime))
+    } catch {
+      case e: Exception =>
+        val LOG = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+        LOG.error(s"Wrong value for carbon.spark.warmUpTime $warmUpTime " +
+                  "Using default Value and proceeding")
+        Thread.sleep(30000)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: default value is 5000, right?

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    val sparkConf = new SparkConf(loadDefaults = true)
+    val builder = SparkSession
+      .builder()
+      .config(sparkConf)
+      .appName("Carbon Thrift Server")
+      .enableHiveSupport()
+
+    val sparkHome = System.getenv.get("SPARK_HOME")
+    if (null != sparkHome) {
+      builder.config("carbon.properties.filepath",
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: In this case, carbon property must be put in SPARK_HOME/conf folder, can user give the path by  spark.driver.extraJavaOptions in spark-default.conf?

Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: No, it seems hard coded now,  It is as per the old thrift server and any changes here require to change another thrift server as well to keep in sync.

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala (hunk 2 above)
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: ok

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala (hunk 1 above)
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/357/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**